sotodlib/tod_ops/flags.py

```diff
@@ -672,3 +692,151 @@ def get_inv_var_flags(aman, signal_name='signal', nsigma=5,
+    """
+    if isinstance(arr, str):
+        arr = aman[arr]
+    if np.isscalar(isub):
+        out = apply_rng(arr, aman.subscan_info.subscan_flags[isub])
+        if trim and out.size == 0:
+            out = None
+    else:
+        if isub is None:
+            isub = range(len(aman.subscan_info.subscan_flags))
+        out = [apply_rng(arr, aman.subscan_info.subscan_flags[ii]) for ii in isub]
+        if trim:
+            out = [x for x in out if x.size > 0]
+
+    return out
+
+
+def apply_rng(arr, rng):
+    """
+    Helper function to get signals on subcsans.
+
+    Parameters
+    ----------
+    arr : Array
+        Array containing the signal. Should have one axis of len (samps)
+    rng : Ranges
+        Ranges object of len (samps) selecting the desired range
+    """
+    if rng.ranges().size == 0:
+        slc = slice(0,0) #  Return an empty array if rng is empty
+    else:
+        slc = slice(*np.squeeze(rng.ranges()))
+    isamps = np.where(np.array(arr.shape) == rng.count)[0][0]
+    ndslice = tuple((slice(None) if ii != isamps else slc for ii in range(arr.ndim)))
+    return arr[ndslice]
+
+def wrap_info(aman, info_aman_name, info, info_names, merge=True):
+    """Wrap info into an aman. Assumed to be (dets,) or (dets, subscans)"""
+    if info[0].ndim == 1:
+        info_aman = core.AxisManager(aman.dets)
+        axmap = [(0, 'dets')]
+    elif info[0].ndim == 2:
+        info_aman = core.AxisManager(aman.dets, aman.subscan_info.subscans)
+        axmap = [(0, 'dets'), (1, 'subscans')]
+
+    for ii in range(len(info_names)):
+        info_aman.wrap(info_names[ii], info[ii], axmap)
+    if merge:
+        if info_aman_name in aman.keys():
+            aman[info_aman_name].merge(info_aman)
+        else:
+            aman.wrap(info_aman_name, info_aman)
+    return info_aman
+
+def get_stats(aman, signal, stat_names, split_subscans=False, mask=None, name="stats", merge=False):
+    """
+    Calculate basic statistics on a TOD or power spectrum.
+
+    Parameters
+    ----------
+    aman : AxisManager
+        Input AxisManager.
+    signal : Array
+        Input signal. Statistics will be computed over *axis 1*.
+    stat_names : list
+        List of strings identifying which statistics to run.
+    split_subscans : bool
+        (Optional). If True statistics will be computed on subscans. Assumes aman.subscans exists already.
+    mask : Array
+        (Optional). Mask to apply before computation. 1d array for advanced indexing, or a slice object.
+    name : str
+        Name of axis manager to add to aman if merge is True.
+    """
+    stat_names = np.atleast_1d(stat_names)
+    fn_dict = {'mean': np.mean, 'median': np.median, 'ptp': np.ptp, 'std': np.std,
+                     'kurtosis': stats.kurtosis, 'skew': stats.skew}
+
+    if isinstance(signal, str):
+        signal = aman[signal]
+    if split_subscans:
+        if mask is not None:
+            raise ValueError("Cannot mask samples and split subscans")
```

Comment:
It should be possible to do a union of the subscan flags + a secondary flag if they're both ranges() aligned along samps. Ok to leave for later but I'd just create an issue for it to track it.

Reply:
Added support for discontinuous ranges to `apply_rng` but didn't implement the full masking. Will add an issue